### DISCUSSION TITLE
How to serve gzip-compressed SVG (.svgz) images on GH Pages?

### DISCUSSION THREAD

**Discussion by Jylpah** [community/community#43528] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello, 

I have a site on GH Pages and I would like to use statically [gzip-compressed SVG images](https://en.wikipedia.org/wiki/SVG#Compression) (`.svgz`) on the site to reduce the repo size. 

**EDIT**. The real issue is GitHub Pages double-compressing SVGZ images, not the `charset=utf-8` HTTP response header. Unless GitHub Pages drops the double-compression from SVGZ images, there is no way to use those on GitHub Pages without external help (e.g. CloudFlare) to rewrite HTTP Request/Response headers. See **[my guide here](https://github.com/Jylpah/svgztest/issues/1)**.

----- Old question for reference ---

I have learned that for a browser to display a `.svgz` image correctly, the response header's `Content-type` field must have both MIME-type and `charset=utf-8` values set: 

```
Content-Type: image/svg+xml; charset=utf-8
```

However, GitHub pages send only the MIME type in response headers' `Content-type` field:

```
Content-Type: image/svg+xml
```

If the `charset=utf-8` value is missing from the response headers (like on Git Hub Pages), browsers (Chrome, Edge, Firefox, Safari) do not display the SVG image. 

Is there a way to enforce the `Content-Type: charset=utf-8` for SVGZ images? **This seems to be a GH Pages server configuration issue for me.**

I use [Hugo](https://gohugo.io/) static site generator, not Jekyll. The repo is [Jylpah/jylpah.github.io](https://github.com/Jylpah/jylpah.github.io). 

**Full response headers:** 

<img width="510" alt="Screenshot 2023-01-07 012326" src="https://user-images.githubusercontent.com/43722525/211117587-9907e87d-94d0-4fd5-ba4a-b051336cec3a.png">

**How it looks on a browsers (Edge):**

<img width="757" alt="Screenshot 2023-01-07 012355" src="https://user-images.githubusercontent.com/43722525/211117703-e7ad2d09-cb57-49e7-b3bf-e26837999407.png">

**Comment by yoannchaudet** [comment]
GitHub Pages relies on https://github.com/jshttp/mime-db for content types. You may want to request a default charset to them.

**Reply by Jylpah** [reply]
Hi @yoannchaudet , I did study this further and it turned out that the real issue was not the missing `charset=utf-8` but GitHub Pages double-compressing SVGZ images. I was thinking to write a PR to the `mime-db`, but since SVG and SVGZ share the same MIME-type, I am not sure how the MIME-type should be specified so that SVG has `"compressible": true`, but SVGZ has `"compressible": false` (assuming GitHub Pages web server follows that). 

I ended up opening a [CloudFlare](https://cloudflare.com/) account just to rewrite HTTP Request/Response headers. Here is a guide 

**[HOWTO serve SVGZ images from GitHub Pages](https://github.com/Jylpah/svgztest/issues/1)**

**Comment by Jylpah** [comment, marked as answer]
**[HOWTO serve SVGZ images from GitHub Pages](https://github.com/Jylpah/svgztest/issues/1)**
